### PR TITLE
DnsClient throws an exception when calling a Dns server over IpV6

### LINKED ISSUE #2456: DNS client should support calling a DNS server over IpV6
Step to reproduce:
`vertx.createDnsClient(53, "2001:4860:4860::8888").lookup4("vertx.io", res -> {});
`
=> the result fails with the following exception: 'java.net.SocketException: Address family not supported by protocol'

Expected: The DnsClient should be able to call a Dns server over IpV4 or IpV6

As a sidenote, I also tried using the full IpV6 address "2001:4860:4860:0:0:0:0:8888" but it doesn't work as well.

If time permits, I'll investigate this issue and provide a fix for it

### PR DESCRIPTION
**Pull request by stephanebastian**
fixes #2456

@vietj could you please review

### CODE REVIEW

**Comment by stephanebastian** [comment]
@vietj I made the change in the updated PR

File: src/test/java/io/vertx/test/core/DNSTest.java [hunk 1]
```diff
@@ -110,6 +110,44 @@ public void testResolveA() throws Exception {
...
+  public void testResolveAIpV6Abbr() throws Exception {
+    // use google servers using ipv6 abbrivated format
+    final String ip = "10.0.0.1";
+    // force the fake dns server to Ipv6
+    DnsClient dns = prepareDns(FakeDNSServer.testResolveA(ip).ipAddress("::1"));
+    dns.resolveA("vertx.io", onSuccess(result -> {
+      assertFalse(result.isEmpty());
+      assertEquals(1, result.size());
+      assertEquals(ip, result.get(0));
+      ((DnsClientImpl) dns).inProgressQueries(num -> {
```
[the remark this answers is not in the document]
Comment (vietj) [reply src/test/java/io/vertx/test/core/DNSTest.java]: can you remove the cast to `DnsClientImpl` ?

[the remark this answers is not in the document]
Comment (stephanebastian) [reply src/test/java/io/vertx/test/core/DNSTest.java]: well, the thing is the interface DnsClient doesn't expose the method inProgressQueries(...).
do you want to remove the whole block that checks if the number of pending queries is zero ?

[the remark this answers is not in the document]
Comment (vietj) [reply src/test/java/io/vertx/test/core/DNSTest.java]: why is it important to keep it ? aren't the three previous statement enough ?

[the remark this answers is not in the document]
Comment (stephanebastian) [reply src/test/java/io/vertx/test/core/DNSTest.java]: it's mainly a cut/paste from existing tests and to be honest I am not sure what it's supposed to test.
Having said that, we've got 2 options here:
- This block of code is mainly useless and we should remove it from *all* tests
```
      ((DnsClientImpl) dns).inProgressQueries(num -> {
        assertEquals(0, (int)num);
        testComplete();
      });
```
- If it's useful we may leave it there.

WDYT ?

**Pull request merged**